taichi/transforms/lower_ast.cpp

```diff
@@ -167,11 +169,67 @@ class LowerAST : public IRVisitor {
       auto end = stmt->end;
       begin->flatten(flattened);
       end->flatten(flattened);
-      auto &&new_for = std::make_unique<RangeForStmt>(
-          stmt->parent->lookup_var(stmt->loop_var_id[0]), begin->stmt,
-          end->stmt, std::move(stmt->body), stmt->vectorize, stmt->parallelize,
-          stmt->block_dim, stmt->strictly_serialized);
-      flattened.push_back(std::move(new_for));
+      bool is_good_range_for = capturing_loop == nullptr || // the outer-most for
+        std::find(detected_fors_with_break.begin(), detected_fors_with_break.end(), stmt)
+        == detected_fors_with_break.end(); // for without break
```

Comment:
Let's add a comment to explain what `good range for` is. `for without break` is slightly vague.